src/endpoints/data_objects/impl.cpp

```diff
@@ -1845,4 +1851,132 @@ namespace
         using namespace irods::http::shared_api_operations;
         return op_atomic_apply_metadata_operations(_sess_ptr, _req, _args, entity_type::data_object);
     } // op_modify_metadata
+
+    IRODS_HTTP_API_ENDPOINT_OPERATION_SIGNATURE(op_modify_replica)
+    {
+        auto result = irods::http::resolve_client_identity(_req);
+        if (result.response) {
+            return _sess_ptr->send(std::move(*result.response));
+        }
+
+        const auto* client_info = result.client_info;
+
+        irods::http::globals::background_task([fn = __func__, client_info, _sess_ptr, _req = std::move(_req), _args = std::move(_args)] {
+            log::info("{}: client_info->username = [{}]", fn, client_info->username);
+
+            http::response<http::string_body> res{http::status::ok, _req.version()};
+            res.set(http::field::server, irods::http::version::server_name);
+            res.set(http::field::content_type, "application/json");
+            res.keep_alive(_req.keep_alive());
+
+            try {
+                DataObjInfo info{};
+                irods::at_scope_exit free_memory{[&info] { clearKeyVal(&info.condInput); }};
+
+                if (const auto iter = _args.find("lpath"); iter != std::end(_args)) {
+                    std::strncpy(info.objPath, iter->second.c_str(), sizeof(DataObjInfo::objPath));
+                }
+                else {
+                    log::error("{}: Missing [lpath] parameter.", fn);
+                    return _sess_ptr->send(irods::http::fail(res, http::status::bad_request));
+                }
+
+                if (auto iter = _args.find("resource-hierarchy"); iter != std::end(_args)) {
+                    std::strncpy(info.rescHier, iter->second.c_str(), sizeof(DataObjInfo::rescHier));
+                }
+                else if (iter = _args.find("replica-number"); iter != std::end(_args)) {
+                    info.replNum = std::stoi(iter->second);
+                }
+                else {
+                    log::error("{}: Missing [resource-hierarchy] or [replica-number] parameter.", fn);
+                    return _sess_ptr->send(irods::http::fail(res, http::status::bad_request));
+                }
+
+                KeyValPair kvp{};
+                irods::at_scope_exit clear_kvp{[&kvp] { clearKeyVal(&kvp); }};
+
+                if (const auto name_iter = _args.find("property-name"); name_iter != std::end(_args)) {
+                    static constexpr auto properties = std::to_array<std::pair<const char*, const char*>>({
+                        {"COLL_ID", COLL_ID_KW},
+                        {"DATA_CREATE_TIME", DATA_CREATE_KW},
+                        {"DATA_CHECKSUM", CHKSUM_KW},
+                        {"DATA_EXPIRY", DATA_EXPIRY_KW},
+                        {"DATA_ID", DATA_ID_KW},
+                        {"DATA_REPL_STATUS", REPL_STATUS_KW},
+                        {"DATA_MAP_ID", DATA_MAP_ID_KW},
```

Comment:
Just noting for the future that DATA_ID and DATA_MAP_ID cannot be modified through the API (enforced by the database plugin).

If we want this to mirror `iadmin modrepl`, we need to disallow modifying the following columns:
 - COLL_ID
 - DATA_ID
 - DATA_MAP_ID
 - DATA_NAME
 - DATA_RESC_HIER
 - DATA_RESC_NAME

Arguably, the API should enforce this, but... I don't think that it does. Also, that would require some admin-only shenanigans as these columns do need to be modified for legitimate cases. Modifying these columns outside of those legitimate cases can really put you in a situation.

I think at the very least that we need to note somewhere that modifying these columns is dangerous and may not be recoverable.

Reply:
Just noting that is not enough I feel.   We need to enforce it - real bumpers.

Feels like those bumpers just being here for now (same as in `iadmin modrepl`) is sufficient to me...

Reply:
Will do.

Reply:
I'm considering the removal of DATA_OWNER_NAME and DATA_OWNER_ZONE. Since ichmod no longer treats them special, it feels right to not allow people to modify them. I don't think there's a valid use-case for allowing those to be modified now.

Reply:
I think i agree.   do we allow those for modrepl?

Reply:
Yes.

Reply:
oh, hmmm.

Reply:
Ok. I'm going to remove them. We can always add them back in if needed.

Reply:
Done.